src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceTest.groovy

```diff
@@ -0,0 +1,199 @@
+        List actualCostOfItems = []
+
+        Cart cartForDiscount = new Cart(UUID.randomUUID(), itemListForDiscount)
+
+        when:
+        orderingService.applyDiscountToCartItems(cartForDiscount);
+        for (Item discountedItem in cartForDiscount.getItems()){
+            actualCostOfItems.add(discountedItem.getCost())
+        }
+
+        then:
+        expectedCostOfItems == actualCostOfItems
+    }
+
+    def "createAnOrder should throw UnableToCreateOrderException if canContainFood is false and cart contains food"() {
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartForOrder = new Cart(UUID.randomUUID(), itemListWithEachType)
+        def canContainFood = false
+
+        when:
+        Order order = orderingService.createAnOrder(cartForOrder, "recipientName",
+                "recipientAddress", canContainFood)
+
+        then:
+        thrown(UnableToCreateOrderException)
+    }
+
+    def "createAnOrder should respond with an object Order with recipient name"() {
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartForOrder = new Cart(UUID.randomUUID(), itemListWithEachType)
+        def canContainFood = true
+        def expectedName = "recipientName"
+        def expectedAddress = "recipientAddress"
+
+        when:
+        Order order = orderingService.createAnOrder(cartForOrder, expectedName, expectedAddress, canContainFood)
+        def actualName = order.getRecipientName()
+
+        then:
+        expectedName == actualName
+    }
+
+    def "createAnOrder should respond with an object Order with recipient address"() {
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartForOrder = new Cart(UUID.randomUUID(), itemListWithEachType)
+        def canContainFood = true
+        def expectedName = "recipientName"
+        def expectedAddress = "recipientAddress"
+
+        when:
+        Order order = orderingService.createAnOrder(cartForOrder, expectedName, expectedAddress, canContainFood)
+        def actualAddress = order.getRecipientAddress()
+
+        then:
+        expectedAddress == actualAddress
+    }
+
+    def "createAnOrder should respond with an object Order with total cost"() {
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartForOrder = new Cart(UUID.randomUUID(), itemListWithEachType)
+        def canContainFood = true
+        def expectedName = "recipientName"
+        def expectedAddress = "recipientAddress"
+        double expectedTotalCost = 1710.0
+
+        when:
+        Order order = orderingService.createAnOrder(cartForOrder, expectedName, expectedAddress, canContainFood)
+        def actualTotalCost = order.getTotalCost()
+
+        then:
+        expectedTotalCost == actualTotalCost
+    }
```

Comment:
Same with this one, we can just have 1 unit test to check all the fields/properties for the returned order.